pkg/types/message_envelope.go

```diff
@@ -38,6 +40,19 @@ type MessageEnvelope struct {
 // NewMessageEnvelope creates a new MessageEnvelope for the specified payload with attributes from the specified context
 func NewMessageEnvelope(payload []byte, ctx context.Context) MessageEnvelope {
 	envelope := MessageEnvelope{
+		Checksum:      "",
+		CorrelationID: fromContext(ctx, correlationId),
+		ContentType:   fromContext(ctx, contentType),
+		Payload:       payload,
+	}
+
+	return envelope
+}
+
+// NewMessageEnvelope creates a new MessageEnvelope for the specified payload with attributes from the specified context along with a Checksum value
+func NewMessageEnvelopeWithChecksum(checksum string, payload []byte, ctx context.Context) MessageEnvelope {
```

Comment:
I have mixed feelings about providing this function. My suggestion would be to just call the original "constructor" function and then set the property on the returned type for the CBOR use case.

Reply:
Agreed. I think this is the best and simplest solution. I will remove this and do as you suggested.